src/compiler/program.ts

```diff
@@ -977,7 +978,15 @@ namespace ts {
         }
 
         function isSourceFileDefaultLibrary(file: SourceFile): boolean {
-            return file.fileName === host.getDefaultLibFileName(options);
+            if (file.hasNoDefaultLib) {
+                return true;
+            }
+
+            if (defaultLibraryPath !== undefined && defaultLibraryPath.length !== 0) {
+                return comparePaths(defaultLibraryPath, file.path, currentDirectory, /*ignoreCase*/ true) === Comparison.EqualTo;

```

Comment:
Do not unconditionally pass `true` for ignoreCase, instead use `!host.useCaseInsensitiveFileNames()`.